src/core/settings.py

```diff
@@ -104,30 +91,54 @@ def _parse_labels(self, labels_str):
             return []
         return [label.strip() for label in labels_str.split(",") if label.strip()]
 
-    def _detect_ai_provider(self):
-        explicit = os.environ.get("INPUT_AI-PROVIDER", "").lower()
-        if explicit in self.providers:
-            if not self.providers[explicit]:
-                raise ValueError(f"{self.providers[explicit]} API key not provided")
-            return explicit
+    def _get_llm_configs(self):
+        gemini_api_key   = os.environ.get("INPUT_GEMINI-API-KEY")
+        openai_api_key   = os.environ.get("INPUT_OPENAI-API-KEY")
+        deepseek_api_key = os.environ.get("INPUT_DEEPSEEK-API-KEY")
+
+        self.ai_providers = {}
+        if gemini_api_key:
+            self.ai_providers["gemini"] = {
+            "provider" : "gemini",
+            "api_key"  : gemini_api_key,
+            "model"    : os.environ.get("INPUT_GEMINI-MODEL")
+        }
+        if openai_api_key:
+            self.ai_providers["openai"] = {
+            "provider" : "openai",
+            "api_key"  : openai_api_key,
+            "model"    : os.environ.get("INPUT_OPENAI-MODEL")
+        }
+        if deepseek_api_key:
+            self.ai_providers["deepseek"] = {
+            "provider" : "deepseek",
+            "api_key"  : deepseek_api_key,
+            "model"    : os.environ.get("INPUT_DEEPSEEK-MODEL")
+        }
 
-        for provider, api_key in self.providers.items():
-            if api_key:
-                return provider
+        if not self.ai_providers:
+            raise ValueError("No LLM API key was provided. Please provide one of the following: deepseek, gemini, openai.")
 
-        raise ValueError(
-            "No LLM API key was provided. Please provide one of the following: deepseek, gemini, openai."
-        )
+        self.explicit_provider = os.environ.get("INPUT_AI-PROVIDER", "").lower()
+        if self.explicit_provider and self.explicit_provider not in self.ai_providers:
+            raise ValueError(f"API key not found for {self.explicit_provider}")
 
-    def get_api_key(self):
-        return self.providers[self.ai_provider]
+        if len(self.ai_providers) == 1:
+            self.explicit_provider = list(self.ai_providers)[0]
+
+        if self.explicit_provider:
+            self.ai_providers = {self.explicit_provider : self.ai_providers[self.explicit_provider]}
+
+    @property
+    def ai_provider(self):
+        ai_provider_name = self.ai_providers[self.explicit_provider] \
+            if self.explicit_provider \
+            else ai_provider = random.choice(list(self.ai_providers.keys())) # noqa: S311
+        return self.ai_providers[ai_provider_name]
```

Comment:
```suggestion
    @property
    def ai_provider(self):
        ai_provider = random.choice(list(self.ai_providers.keys()))
        return self.ai_providers[ai_provider]
```
WDYT about this solution? Yes, it might call random.choice a few more times, even when there's only one element, but code looks simple